### PR TITLE
helm: Optionally set userID and userKey in cephfs chart.

### PR DESCRIPTION
**Pull request by james-choncholas**
The cephfs static provisioner expect userID and userKey in the credential secret. This PR adds these values to the helm chart so that they are only included in the templated yaml if the values are non-empty.

# Describe what this PR does #

This PR adds userID and userKey to the cephfs provisioner helm chart. These credentials are required for static provisioning. The userID and userKey are optional values in values.yaml. If they are not provided, they are excluded from the secret.

Is the change backward compatible? Yes

Are there concerns around backward compatibility? No

## Related issues ##

https://github.com/ceph/ceph-csi/issues/4467

## Future concerns ##

None

**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow
  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [x] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [x] [Pending release
  notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [x] Documentation has been updated, if necessary.
* [ ] Unit tests have been added, if necessary.
* [ ] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

### CODE REVIEW

File: charts/ceph-csi-cephfs/values.yaml [hunk 1]
```diff
@@ -331,6 +331,9 @@ secret:
   # specified in the storage class
   adminID: <plaintext ID>
   adminKey: <Ceph auth key corresponding to ID above>
+  # User credentials are required for the static provisioner.
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/values.yaml]:
```suggestion
  # User credentials are required for the static provisioned PVC.
```
can you please update the same above?

**Review by Madhu-1 — COMMENTED** [review]
small nit

**Comment by Rakshith-R** [comment]
Please check commitlint failure as well.
https://github.com/ceph/ceph-csi/actions/runs/10550648436/job/29243552546?pr=4801

File: charts/ceph-csi-cephfs/values.yaml (hunk 1 above)
Comment (james-choncholas) [reply charts/ceph-csi-cephfs/values.yaml]: Sure, I fixed this occurrence and the two others in the chart README.

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Comment by Madhu-1** [comment]
@mergifyio queue

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Comment by Madhu-1** [comment]
@Mergifyio queue

**Pull request merged**